### PR TITLE
Throw a proper error when topology file initialization failed

### PR DESCRIPTION
**Pull request by Infinisil**
In particular when the file is missing or contains invalid json

### CODE REVIEW

File: morpho-checkpoint-node/src/Morpho/Node/Env.hs [hunk 1]
```diff
@@ -113,14 +113,17 @@ initProducers :: NodeConfiguration -> IO [RemoteAddress]
...
-
-  case List.lookup nid $
-    map (\ns -> (CoreNodeId $ nodeId ns, producers ns)) topology of
-    Just ps -> return ps
-    Nothing -> do
-      hPutStrLn stderr $ "Node " <> Text.pack (show nid) <> " not found in topology file " <> Text.pack (show file)
+  mtopology <- readTopologyFile file
+  case mtopology of
+    Left err -> do
+      hPutStrLn stderr $ Text.pack err
```
Comment (koslambrou) [inline morpho-checkpoint-node/src/Morpho/Node/Env.hs:119]: When there's an error in the JSON file, the error is:

```
Error in $: not enough input. Expecting ',' or ']'
```

What is that dollar sign?

**Review by koslambrou — CHANGES_REQUESTED** [review]

File: morpho-checkpoint-node/src/Morpho/Node/Env.hs (hunk 1 above)
Comment (Infinisil) [reply morpho-checkpoint-node/src/Morpho/Node/Env.hs:119]: This apparently represents the root of a JSON path, see `Data.Aeson.Types.formatPath`. There doesn't appear to be an easy way to format it better

Comment (koslambrou) [reply morpho-checkpoint-node/src/Morpho/Node/Env.hs:119]: Maybe say that the error is in the topology file?

Comment (Infinisil) [reply morpho-checkpoint-node/src/Morpho/Node/Env.hs:119]: Oh yeah good point, I'll fix that

**Comment by Infinisil** [comment]
Now when the topology file is missing:
```
Failed to read topology file configuration/morpho-topology.json: configuration/morpho-topology.json: openBinaryFile: does not exist (No such file or directory)
```

And when it can't be decoded:
```
Failed to decode topology file configuration/morpho-topology.json: Error in $: not enough input. Expecting ',' or ']'
```

File: morpho-checkpoint-node/src/Morpho/Node/Env.hs (hunk 1 above)
Comment (Infinisil) [reply morpho-checkpoint-node/src/Morpho/Node/Env.hs:119]: Done that now

**Review by koslambrou — APPROVED** [review]

**Pull request merged**